src/client.ts

```diff
@@ -1339,17 +1342,46 @@ export class StreamChat<
     }
 
     /** update the client.state with any changes to users */
-    if (event.type === 'user.presence.changed' || event.type === 'user.updated') {
-      if (event.user?.id === this.userID) {
-        this.user = this.user && { ...this.user, ...event.user };
-        /** Updating only available properties in _user object. */
-        Object.keys(event.user).forEach((key) => {
-          if (this._user && key in this._user) {
-            /** @ts-expect-error */
-            this._user[key] = event.user[key];
-          }
-        });
+    if (
+      (event.type === 'user.presence.changed' || event.type === 'user.updated') &&
+      event.user.id === this.userID
+    ) {
+      const ownUserProperties = [
+        'channel_mutes',
+        'devices',
+        'mutes',
+        'total_unread_count',
+        'unread_channels',
+        'unread_count',
+        'invisible',
+        'roles',
+      ];
+
+      // Remove deleted properties from user objects.
+      for (const key in this.user) {
+        if (key in event.user || ownUserProperties.includes(key)) {
```

Comment:
why do we care about the ownUserProperties here?

Reply:
We don't want to remove those ownUserProperties ... they are kinda reserved fields!!

Reply:
oh i see and they are not in the event object, in that case maybe we should extract these fields and put them somewhere as a source of truth, in future, if we add more reserve fields and forgot to add them here it will cause hard to find issues